### PR TITLE
Keep input dimensions consisten

### PR DESCRIPTION
**Pull request by wiederm**
## Description
Input dimensions for `inputs["atomic_numbers"]` has shape (nr_of_atoms_in_batch, 1), which was flattended in the `BaseNNP` before. Now we keep dimensions and let the NNP decide what to do with it.

## Question
Should we change the name of the input property `atomic_number` 
https://github.com/choderalab/modelforge/blob/6dab8a1a376a8b0cf83d84ac927793154239e272/modelforge/potential/models.py#L172
 to `atomic_properties`? The way we are approaching the embedding in this PR makes it flexible to have other properties defined in this tensor with the now flexible shape `(nr_of_atoms_in_batch, *, *)`. What do you think @ArnNag ?

## Status
- [x] Ready to go

### CODE REVIEW

File: modelforge/potential/utils.py [hunk 1]
```diff
@@ -5,6 +5,48 @@
 import torch.nn as nn
 
 
+class SlicedEmbedding(nn.Module):
+    def __init__(
```
Comment (chrisiacovella) [inline modelforge/potential/utils.py:9]: Can you add in a docstring for this class?

File: modelforge/tests/helper_functions.py [hunk 2]
```diff
@@ -38,8 +38,10 @@ def setup_simple_model(
         Initialized model.
     """
     from modelforge.potential import CosineCutoff, GaussianRBF
+    from modelforge.potential.utils import SlicedEmbedding
 
-    embedding = torch.nn.Embedding(max_atomic_number, nr_atom_basis)
+    embedding = SlicedEmbedding(max_atomic_number, nr_atom_basis, sliced_dim=0)
+    assert embedding.embedding_dim == nr_atom_basis
```
Comment (chrisiacovella) [inline modelforge/tests/helper_functions.py:44]: There doesn't seem to really be a test of the functionality of the `SlicedEmbedding` class.

**Review by chrisiacovella — CHANGES_REQUESTED** [review]
Not too much to do on this, but I think there needs to be some tests of `SlicedEmbedding` class.  right now it is only checking the dimension that gets set in the class.

File: modelforge/potential/utils.py (hunk 1 above)
Comment (wiederm) [reply modelforge/potential/utils.py:9]: sure!

**Review by chrisiacovella — APPROVED** [review]
Looks good, lets get this merged!

**Pull request merged**